packages/components/src/base-control/index.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import type { FunctionComponent } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { VisuallyHidden } from '../visually-hidden';
+import type { BaseControlProps, BaseControlVisualLabelProps } from './types';
+import {
+	Wrapper,
+	StyledField,
+	StyledLabel,
+	StyledHelp,
+	StyledVisualLabel,
+} from './styles/base-control-styles';
+
+/**
+ * `BaseControl` is a component used to generate labels and help text for components handling user inputs.
+ *
+ * @example
+ * // Render a `BaseControl` for a textarea input
+ * import { BaseControl } from '@wordpress/components';
+ *
+ * // The `id` prop is necessary to accessibly associate the label with the textarea
+ * const MyBaseControl = () => (
+ *   <BaseControl id="textarea-1" label="Text" help="Enter some text" __nextHasNoMarginBottom={ true }>
+ *     <textarea id="textarea-1" />
+ *   </BaseControl>
+ * );
+ */
+export const BaseControl: FunctionComponent< BaseControlProps > & {
+	VisualLabel: typeof VisualLabel;
+} = ( {
+	__nextHasNoMarginBottom = false,
+	id,
+	label,
+	hideLabelFromVision = false,
+	help,
+	className,
+	children,
+} ) => {
+	return (
+		<Wrapper
+			className={ classnames( 'components-base-control', className ) }
+		>
+			<StyledField
+				className="components-base-control__field"
+				// TODO: Official deprecation for this should start after all internal usages have been migrated
+				__nextHasNoMarginBottom={ __nextHasNoMarginBottom }
+			>
+				{ label &&
+					id &&
+					( hideLabelFromVision ? (
```

Comment:
I know that these lines of code were not touched in this PR, but I find the logic here hard unintuitive to read